src/utils/work/seekWorkResponse.js

```diff
@@ -176,8 +176,67 @@ const dispatchWorkRequest = async (
     },
   );
 
-  // TODO switch to using normal WorkRequest for v2 requests
   return await Promise.race([timeoutPromise, responsePromise]);
 };
 
-export { dispatchWorkRequest, seekFromS3 };
+const seekWorkerResultsOrDispatchWork = async (
```

Comment:
this function is too long in my opinion and the name (although long) doesn't explain everything it does because it also deals with the cache. I would
1.  rename it to something simpler like `getWorkData` or `getWorkRequestData` (remember to rename the file not only the function because that's the style we try to follow: function exported = file name)
2. Split the 3 different parts of the function into smaller functions: function dealing with cache, signed url, and dispatch work request.

This is an important and very used flow so I'd like to be able to have a quick look and see a high-level function where you see that first you try cache, then signed url and then the work request, without `if`s and error handling in between.

Reply:
if you have another suggestion on making this a bit cleaner it's ok too, after checking more in detail the function is not that complex but imo it's harder to read than it needs.